File: java/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1, area = 0;
        while (left < right) {
            area = Math.max(area, Math.min(height[left], height[right]) * (right - left));
            if (height[right] > height[left])
                left++;
            else
                right--;
        }
        return area;
    }
}

File: cpp/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int area = INT_MIN;
        int left = 0, right = height.size() - 1;
        while (left < right) {
            area = max(area, min(height[right], height[left]) * (right - left));
            height[right] > height[left] ? left++ : right--;
        }
        return area;
    }
};
